File: Kotlin/Quiz.kt
import java.util.*

data class QuestionAnswer (
    val q: String,
    val a: String
)

fun main() {

    var questions = arrayOf(
        QuestionAnswer("In what year was Kotlin first released?", "2011"),
        QuestionAnswer("In what year was Kotlin 1.0 released?", "2016"),
        QuestionAnswer("What organisation is responsible for Kotlin?", "JetBrains"),
        QuestionAnswer("Who was the lead designer of Kotlin?", "Andrey Breslav"),
        QuestionAnswer("What language was Kotlin intended to be 'better than'?", "Java")
    );

    var incorrectGuesses = 0

    for (question in questions) {
        println(question.q)

        var guess = ""
        while (guess != question.a && guess != "skip") {
            print("> ")
            guess = readLine()!!

            if (guess == "skip") {
                incorrectGuesses += 10
                println(question.a)
                println()
            } else if (guess != question.a) {
                incorrectGuesses++
            } else {
                println("Correct")
                println()
            }
        }
    }

    println("Incorrect Guesses: " + incorrectGuesses.toString())
}

File: Swift/quiz.swift

struct QuestionAnswer {
    var q = ""
    var a = ""
}

func main() {
    let questions = [
        QuestionAnswer(q: "In what year was Swift first released?", a: "2014"),
        QuestionAnswer(q: "Which individual is credited with the creation of Swift?", a: "Chris Lattner"),
        QuestionAnswer(q: "What organisation is responsible for Swift?", a: "Apple"),
        QuestionAnswer(q: "What language was Swift intended to replace?", a: "Objective-C"),
    ]

    var incorrectGuesses = 0

    questions.forEach { question in
        print(question.q)

        var guess = ""
        while guess != question.a && guess != "skip" {
            print("> ", terminator: "")
            guess = readLine() ?? ""

            if guess == "skip" {
                incorrectGuesses += 10
                print(question.a)
                print()
            } else if guess != question.a {
                incorrectGuesses += 1
            } else {
                print("Correct")
                print()
            }
        }
    }

    print("Incorrect Guesses: " + String(incorrectGuesses))
}
main()
